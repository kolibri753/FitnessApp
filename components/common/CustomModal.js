import React from "react";
import { View, Text, StyleSheet, TouchableOpacity, Modal } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import { colors } from "../../styles/colors";

const CustomModal = ({ isModalVisible, closeModal, title, children }) => {
	return (
		<Modal animationType="slide" transparent={true} visible={isModalVisible}>
			<View style={styles.modalContainer}>
				<View style={styles.modalContent}>
					<TouchableOpacity
						style={styles.closeButton}
						onPress={closeModal}
						testID="closeButton"
					>
						<MaterialIcons name="close" size={24} color={colors.black} />
					</TouchableOpacity>
					<Text style={styles.modalTitle}>{title}</Text>
					{children}
				</View>
			</View>
		</Modal>
	);
};

const styles = StyleSheet.create({
	modalContainer: {
		flex: 1,
		justifyContent: "center",
		alignItems: "center",
		backgroundColor: colors.grey,
	},
	modalContent: {
		backgroundColor: colors.lightGrey,
		borderRadius: 10,
		padding: 20,
		width: "90%",
		alignItems: "center",
	},
	closeButton: {
		alignSelf: "flex-end",
	},
	modalTitle: {
		fontSize: 18,
		fontWeight: "bold",
		marginTop: 10,
		marginBottom: 20,
	},
});

export default CustomModal;
