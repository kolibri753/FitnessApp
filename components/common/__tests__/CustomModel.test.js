import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import CustomModal from "../CustomModal";

describe("CustomModal", () => {
	const onCloseModal = jest.fn();
	const title = "Test Modal";
	const children = <Text>Test Content</Text>;

	it("renders correctly", () => {
		const { getByText } = render(
			<CustomModal isModalVisible={true} closeModal={onCloseModal} title={title}>
				{children}
			</CustomModal>
		);

		expect(getByText(title)).toBeTruthy();
		expect(getByText("Test Content")).toBeTruthy();
	});

	it("calls closeModal function when close button is pressed", () => {
		const { getByTestId } = render(
			<CustomModal isModalVisible={true} closeModal={onCloseModal} title={title}>
				{children}
			</CustomModal>
		);

		fireEvent.press(getByTestId("closeButton"));
		expect(onCloseModal).toHaveBeenCalledTimes(1);
	});

	it("renders correctly", () => {
		const tree = renderer.create(<CustomModal />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
