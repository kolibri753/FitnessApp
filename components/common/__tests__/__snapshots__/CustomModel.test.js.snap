// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#333333",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#eee",
          "borderRadius": 10,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
          }
        }
        testID="closeButton"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "material",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          Óóç
        </Text>
      </View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginTop": 10,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
